pkg/core/native/designate.go

```diff
@@ -95,9 +95,8 @@ func copyDesignationCache(src, dst *DesignationCache) {
 	*dst = *src
 }
 
-func (s *Designate) isValidRole(r noderoles.Role) bool {
-	return r == noderoles.Oracle || r == noderoles.StateValidator ||
-		r == noderoles.NeoFSAlphabet || r == noderoles.P2PNotary
+func (*Designate) isValidRole(r noderoles.Role) bool {
+	return noderoles.IsValid(r)
```

Comment:
Just replace all usages of `(*Designate) isValidRole` with `noderoles.IsValid` and remove `(*Designate) isValidRole`.